File: examples/a2a-backend/python/account_agent.py
from light_a2a_backend import AdapterConfig, ReplayStore, serve
from pathlib import Path

class AccountAgent:
    def capabilities(self): return {"contractVersion":"light-a2a-backend/v1","streaming":False,"cancellation":True,"statusReconciliation":True,"acceptedContentModes":["text/plain"],"maximumArtifactBytes":1048576}
    def invoke(self, _context, request): return {"state":"COMPLETED","backendOperationId":"account:"+request["taskId"],"result":{"answer":"business result"},"error":None,"artifacts":[]}
    def invoke_stream(self, _context, _request): return iter(())
    def status(self, _context, request): return self.invoke(_context,request)
    def cancel(self, _context, request): return {"state":"CANCELED","backendOperationId":"account:"+request["taskId"],"result":None,"error":None,"artifacts":[]}

# Deployment supplies AdapterConfig from its generated business-agent config;
# business code above has no HTTP, signature, replay, token, or Portal logic.

File: examples/a2a-backend/typescript/account-agent.js
export const accountAgent = {
  capabilities: () => ({ contractVersion: "light-a2a-backend/v1", streaming: false, cancellation: true, statusReconciliation: true, acceptedContentModes: ["text/plain"], maximumArtifactBytes: 1_048_576 }),
  invoke: async (_context, request) => ({ state: "COMPLETED", backendOperationId: `account:${request.taskId}`, result: { answer: "business result" }, error: null, artifacts: [] }),
  invokeStream: async function* () {},
  status: async (context, request) => accountAgent.invoke(context, request),
  cancel: async (_context, request) => ({ state: "CANCELED", backendOperationId: `account:${request.taskId}`, result: null, error: null, artifacts: [] }),
};

// Deployment calls the SDK serve() function with generated adapter config.
// This file intentionally contains business callbacks only.
